x-pack/plugins/canvas/canvas_plugin_src/renderers/embeddable/embeddable_input_to_expression.ts

```diff
@@ -21,17 +21,24 @@ export const inputToExpressionTypeMap = {
 /*
   Take the input from an embeddable and the type of embeddable and convert it into an expression
 */
-export function embeddableInputToExpression(
-  input: EmbeddableInput,
+export function embeddableInputToExpression<
+  B extends boolean,
+  C = B extends true ? string : string | undefined
+>(
```

Comment:
This might be an over-complicated type - but basically, I wanted to ensure that **when `useGenericEmbeddable` is true**, Typescript knows that the function will always return a string and not undefined. This avoids having to add `ts-ignore` comments wherever I added a call to `embeddableInputToExpression`

Reply:
Good call - renamed `B` and `C` (was meant to be `boolean` and `conditional` LOL but definitely not clear) in e98aef7d72b3881282d025108f8540a4460bb978 :+1: